File: src/lib/logger.ts
import { invoke } from "@tauri-apps/api/core";

const isTauri = () => typeof window !== "undefined" && ("__TAURI_INTERNALS__" in window);

async function log(cmd: string, message: string, context?: any) {
  console.log(`[${cmd}]`, message, context ?? "");

  if (!isTauri()) return;

  try {
    await invoke(cmd, { message, context: context ?? null });
  } catch {
  }
}

export const logger = {
  debug: (message: string, context?: any) => log("log_debug", message, context),
  info: (message: string, context?: any) => log("log_info", message, context),
  warn: (message: string, context?: any) => log("log_warn", message, context),
  error: (message: string, context?: any) => log("log_error", message, context),
};

File: src-tauri/src/utils/logger.rs


#[tauri::command]
fn log_debug(message: String) {
    tracing::debug!("[React] {}", message);
}

#[tauri::command]
fn log_info(message: String) {
    tracing::info!("[React] {}", message);
}

#[tauri::command]
fn log_warn(message: String) {
    tracing::warn!("[React] {}", message);
}

#[tauri::command]
fn log_error(message: String) {
    tracing::error!("[React] {}", message);
}